code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java

```diff
@@ -262,7 +262,15 @@ private void processMonitorRequestEvent(final Event event) {
 		if (PlacesMonitorConstants.EVENTNAME_START.equals(eventName)) {
 			startMonitoring();
 		} else if (PlacesMonitorConstants.EVENTNAME_STOP.equals(eventName)) {
-			stopMonitoring();
+
+			EventData data = event.getData();
+			if(data != null && !data.isEmpty()){
+				boolean shouldClear = data.optBoolean(PlacesMonitorConstants.EventDataKeys.EVENT_DATA_CLEAR, false);
+				stopMonitoring(shouldClear);
+				return;
+			}
+
+			stopMonitoring(false);
```

Comment:
if an stop event is dispatched with empty eventData then the places states wont be cleared